### PR TITLE
Packages n declare fixes

### PR DESCRIPTION
**Pull request by marcmengel**
make_packages_yaml fixes, one other small change.

### CODE REVIEW

**Comment by marcmengel** [comment]
This should get Reallly Close to your hand-built packages.yaml

File: bin/declare_simple [hunk 1]
```diff
@@ -162,5 +156,9 @@ def make_spec(name, version):
     dbf.close()
 
     run_command("spack reindex")
+    #run_command("spack module ups_table refresh")
+    #run_command("spack module ups_version refresh")
+    #run_command("spack module tcl refresh")
+    #run_command("spack module lmod refresh")
```
Comment (greenc-FNAL) [inline bin/declare_simple:162]: Redundant, or diagnostic disables that should have been re-enabled?

File: bin/make_packages_yaml [hunk 2]
```diff
@@ -53,11 +61,16 @@ Linux)
         if [ x$RPMCACHE != x ]
         then
         getv() {
-           grep "^$1-[0-9]" $RPMCACHE | tail -1 | grep -v 'is not installed' | sed -e 's/[^-]*-//' -e 's/[a-z][^-]*-//' -e 's/-.*//' || runversion $1
+           grep "^$1-[0-9]" $RPMCACHE | tail -1 | grep -v 'is not installed' | 
+perl-Data-Dumper-2.145-3.el7.x86_64
+              sed -e 's/.*-\([0-9][0-9.kp]*\)-[0-9][0-9.]*\..*/\1/' ||
+              runversion $1
```
Comment (greenc-FNAL) [inline bin/make_packages_yaml]: Not sure what's going on here?

File: templates/packagelist [hunk 3]
```diff
@@ -100,13 +106,14 @@ tcl:tcl:tcl-devel
...
-xorg-x11-font-utils:mkfontscale:xorg-x11-font-utils
 xorg-x11-server-Xorg:xorg-server:xorg-x11-server-Xorg
+xorg-x11-proto-devel:xextproto:xorg-x11-proto-devel
+xorg-x11-proto-devel:xproto:xorg-x11-proto-devel
 xorg-x11-xtrans-devel:xtrans:xorg-x11-xtrans-devel
 xxhash:xxhash:xxhash-devel
 xz:xz:xz-devel
 z3:z3:z3-devel
 zlib:zlib:zlib-devel
+zstd:zstd:zstd
```
Comment (greenc-FNAL) [inline templates/packagelist]: Missing `<newline>`?

**Review by greenc-FNAL — CHANGES_REQUESTED** [review]

File: bin/make_packages_yaml (hunk 2 above)
Comment (greenc-FNAL) [reply bin/make_packages_yaml]: `perl-DataDumper-...` is a package spec, but you're treating it as a command and then breaking the pipe chain? Looks like an erroneous paste...

Comment (marcmengel) [reply bin/make_packages_yaml]: This is what I get for trying to resolve merges in the github GUI... sigh.

Comment (marcmengel) [reply bin/make_packages_yaml]: fixed.

File: bin/declare_simple (hunk 1 above)
Comment (marcmengel) [reply bin/declare_simple:162]: I was considering adding these until I tried them.  It is exceedingly slow.   I need to find the hook to call to generate one packages module files that install calls...

**Review by greenc-FNAL — APPROVED** [review]

**Pull request merged**